### PR TITLE
feat(medusa): "Gracefully" handle payment session errors

### PR DESCRIPTION
**Pull request by adrien2p**
**What**

The current behavior of the cart setPaymentSessions is to throw and revert all payment sessions in the case where one of the create fails. This PR introduces the possibility of still creating the payment session that succeeds and ignoring the one that fails. The failed creation are logged for the administrator to be informed and aware of the issue. The pros of this are that the customer is no longer blocked during is payment process and only the succeeded payment sessions can be chosen by the customer.

FIXES CORE-669

### CODE REVIEW

File: packages/medusa/src/services/cart.ts
Comment (olivermrbl) [inline packages/medusa/src/services/cart.ts:1816]: comment: Ideally, we would also gracefully handle sessions that fail to update. I realise the case it a bit more tricky, as it could lead to stale session data. Have you thought of a way of handling this? One way would be to delete the session entirely.

Comment (adrien2p) [reply packages/medusa/src/services/cart.ts:1816]: I did not want to touch the update for this exact reason. We can discuss it

File: packages/medusa/src/services/cart.ts
Comment (olivermrbl) [inline packages/medusa/src/services/cart.ts:1789]: praise: I believe this is the best way of handling errors caused by third-party providers. Does not block the checkout flow for the end customer and allows the server admin to pick up the issue.

What do you think @srindom?

**Review by olivermrbl — APPROVED** [review]
LGTM for the first iteration.

Long term, we should think of a pattern allowing us to provide helpful error messages from third-party providers (not only relevant for payment plugins) in the API response.

Pending approval from @srindom.

**Comment by olivermrbl** [comment]
@srindom Do you have anything to add on this one? If not, I'd like to merge to avoid it blocking progress

**Comment by olivermrbl** [comment]
@adrien2p Can I get you to resolve conflicts?

**Comment by olivermrbl** [comment]
@adrien2p Can I get you to resolve conflicts?

**Comment by srindom** [comment]
LGTM!

**Comment by olivermrbl** [comment]
On hold for now, as the proposal to revamp the payment session flow makes this irrelevant.

**Pull request closed without merging**
